core/embed/extmod/modtrezorcrypto/modtrezorcrypto-random.h

```diff
@@ -40,22 +44,52 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorcrypto_random_uniform_obj,
                                  mod_trezorcrypto_random_uniform);
 
 /// import builtins
-/// def bytes(len: int) -> builtins.bytes:
+/// def bytes(len: int, strong: bool = False) -> builtins.bytes:
 ///     """
-///     Generate random bytes sequence of length len.
+///     Generate random bytes sequence of length len. If `strong` is set then
+///     maximum sources of entropy are used.
 ///     """
-STATIC mp_obj_t mod_trezorcrypto_random_bytes(mp_obj_t len) {
-  uint32_t l = trezor_obj_get_uint(len);
-  if (l > 1024) {
+STATIC mp_obj_t mod_trezorcrypto_random_bytes(size_t n_args,
+                                              const mp_obj_t *args) {
+  uint32_t len = trezor_obj_get_uint(args[0]);
+  if (len > 1024) {
     mp_raise_ValueError("Maximum requested size is 1024");
   }
   vstr_t vstr = {0};
-  vstr_init_len(&vstr, l);
-  random_buffer((uint8_t *)vstr.buf, l);
+  vstr_init_len(&vstr, len);
+#if USE_OPTIGA
+  if (n_args > 1 && args[1] == mp_const_true) {
+    uint8_t *dest = (uint8_t *)vstr.buf;
+    if (!optiga_random_buffer(dest, len)) {
+      vstr_clear(&vstr);
+      mp_raise_msg(&mp_type_RuntimeError,
+                   "Failed to get randomness from Optiga.");
+    }
```

Comment:
Should we fail to generate a seed if Optiga fails to generate randomness or should we silently ignore? I think it's better to err on the side of caution and fail. We could also retry, but if something is wrong with the communication it's probably best for the user to replug the device.